ext/storage_sources/s3_store/README.md

```diff
@@ -0,0 +1,55 @@
+# S3 Store Extension
+
+## Building and running
```

Comment:
We can restructure a little bit. From reading the text it is not evident that there are two methods to build:
1. Meeting AWS dependency through system installed library
2. Letting CMake manage the AWS dependency as an external project, letting it download and compile each time.

Then go through the steps for each.

You have covered that information here, but if we present it in the above form it will be clearer to someone new